pkg/controller/jenkins/jenkins_controller.go

```diff
@@ -151,11 +144,32 @@ func (r *ReconcileJenkins) Reconcile(request reconcile.Request) (reconcile.Resul
 			}
 		}
 		reconcileErrors[request.Name] = lastErrors
-		if lastErrors.counter >= 15 {
+		if lastErrors.counter >= reconcileFailLimit {
 			if log.Debug {
-				logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed ten times with the same error, giving up: %+v", err))
+				logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed %d times with the same error, giving up: %+v", reconcileFailLimit, err))
 			} else {
-				logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed ten times with the same error, giving up: %s", err))
+				logger.V(log.VWarn).Info(fmt.Sprintf("Reconcile loop failed %d times with the same error, giving up: %s", reconcileFailLimit, err))
+			}
+
+			jenkins := &v1alpha2.Jenkins{}
+			err := r.client.Get(context.TODO(), request.NamespacedName, jenkins)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// Request object not found, could have been deleted after reconcile request.
+					// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+					// Return and don't requeue
+					return reconcile.Result{}, nil
+				}
+				// Error reading the object - requeue the request.
+				return reconcile.Result{}, errors.WithStack(err)
+			}
```

Comment:
result, err := r.reconcile(request, logger) should return jenkins
don't fetch jenkins again here